### DISCUSSION TITLE
Does pure ftpd need root privileges?

### DISCUSSION THREAD

**Discussion by XIYO** [Homebrew/discussions#1351] [Everyday usage] [answered]
I used a translator because I'm not good at English.
I'm thinking about 3 days because of this problem.

I have installed pure-ftpd on Mac Catalina.
However, I was unable to connect to ftpd locally.
Anonymous, which was activated as the default option, was also unable to connect, and it was impossible to use my Mac account as it is.
I have read all the manuals of pure-ftpd but couldn't find the right answer.
(The cause of the problem is (the service is activated), but the login information cannot be read properly.
The anonymous setting also doesn't work.)

```
The last method I tried was successful. (External connection is also successful)

1. vim /etc/pam.d (https://download.pureftpd.org/pub/pure-ftpd/doc/README.MacOS-X)
2. sudo /usr/local/Cellar/pure-ftpd/x.x.x/sbin/pure-ftpd
```

In the manual
It says you just need to set /usr/local/etc/pure-ftpd.conf, but it doesn't seem to work at all. (I enabled the ability to show the contents of fortune every time I access it, but it doesn't work at all.
It only works by using sudo /usr/local/Cellar/pure-ftpd/x.x.x/sbin/pure-ftpd -f xxx/xxx/zippy. )

Shouldn't you have to edit /usr/local/etc/pure-ftpd.conf? I'm not really sure what to do.

> To sum up my question is:
> 
> 1. I want to connect to ftp without using sudo.
> 2. Why is the modification of pure-ftpd.conf not reflected? (Is it overridden by the plist's argument?)

Please let me know if you are in the wrong place to ask a question. I'll fix it.

Save me from hell...

**Comment by vincentBill** [comment, marked as answer]
Forget Pure-ftpd for macos.
It is running but accept no connection, no log written if a client try to connect, there is no support and nobody use it recently (all forum contributions are very old (>10years).
I have spend 2 days to try to run if on Macos Catalina or Monterey without success.
